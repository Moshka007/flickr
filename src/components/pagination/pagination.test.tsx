import React from "react";
import { render, screen } from "@testing-library/react";
import Pages from "./pagination";
import userEvent from "@testing-library/user-event";


let page: number = 1;
function setPage(pageSetValue: number): void {
    page = pageSetValue;
};

describe('Testing <Pages/>', () => {
    it('Pages have rendered correctly', () => {
        const pages = render(<Pages page={page} setPage={setPage}/>);
        expect(pages).toMatchSnapshot();
    });

    it('Pages have buttons: <Forward, Next>', () => {
        render(<Pages page={page} setPage={setPage}/>)
        expect(screen.queryByText(/< back/i)).toBeInTheDocument();
        expect(screen.queryByText(/forward >/i)).toBeInTheDocument();
    })
})

describe('Testing state pages', () => {
    it('onClick "< back" working correctly', () => {
        render(<Pages page={page} setPage={setPage}/>)

        userEvent.click(screen.getByText(/< back/i));
        expect(page).toEqual(1);
    })

    it('onClick "forward >" working correctly', () => {
        render(<Pages page={page} setPage={setPage}/>)

        userEvent.click(screen.getByText(/Next/i));
        expect(page).toEqual(1);
    })

})
