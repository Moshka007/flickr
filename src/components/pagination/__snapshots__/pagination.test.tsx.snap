// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Pages/> Pages have rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="div"
      >
        <ul
          class="ml-auto pagination"
          style="font-size: 0.85rem;"
        >
          <li
            class="page-item"
          >
            <a
              class="page-link"
              href="#"
              role="button"
            >
              <span
                aria-hidden="true"
              >
                &lt;
                 Back
              </span>
              <span
                class="sr-only"
              >
                Previous
              </span>
            </a>
          </li>
          <li
            class="page-item"
          >
            <a
              class="page-link"
              href="#"
              role="button"
            >
              Page 
              1
               of 
              0
            </a>
          </li>
          <li
            class="page-item"
          >
            <a
              class="page-link"
              href="#"
              role="button"
            >
              <span
                aria-hidden="true"
              >
                Forward 
                &gt;
              </span>
              <span
                class="sr-only"
              >
                Next
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="div"
    >
      <ul
        class="ml-auto pagination"
        style="font-size: 0.85rem;"
      >
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            <span
              aria-hidden="true"
            >
              &lt;
               Back
            </span>
            <span
              class="sr-only"
            >
              Previous
            </span>
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            Page 
            1
             of 
            0
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            <span
              aria-hidden="true"
            >
              Forward 
              &gt;
            </span>
            <span
              class="sr-only"
            >
              Next
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
